packages/edit-site/src/components/global-styles/style-variations-container.js

```diff
@@ -41,10 +44,32 @@ function Variation( { variation } ) {
 	}, [ variation, base ] );
 
 	const selectVariation = () => {
+		const blockStyles = variation?.styles?.blocks || {};
+		if ( user?.styles?.blocks && preserveAdditionalCSS ) {
+			Object.keys( user.styles.blocks ).forEach( ( blockName ) => {
+				if ( user.styles.blocks[ blockName ].css ) {
+					blockStyles[ blockName ] = {
+						...( blockStyles[ blockName ]
+							? blockStyles[ blockName ]
+							: {} ),
+						css: user.styles.blocks[ blockName ].css,
```

Comment:
need to modify this as theme variations can have block level CSS set so we need to merge instead of overwriting here